Dockerfile

```diff
@@ -0,0 +1,45 @@
+# This file is for use as a devcontainer and a runtime container
+# 
+# The devcontainer should use the build target and run as root with podman 
+# or docker with user namespaces.
+#
+FROM python:3.10 as build
+
+# Add any system dependencies for the developer/build environment here
+RUN apt-get update && apt-get upgrade -y && \
+    apt-get install -y --no-install-recommends \
+    build-essential \
+    busybox \
+    git \
+    net-tools \
+    vim \
+    && rm -rf /var/lib/apt/lists/* \
+    && busybox --install
+
+COPY . /project
+
+RUN cd /project && \
+    pip install --upgrade pip build && \
+    export SOURCE_DATE_EPOCH=$(git log -1 --pretty=%ct) && \
+    python -m build --sdist --wheel && \
+    touch requirements.txt
+
+RUN python -m venv /venv
+ENV PATH=/venv/bin:$PATH
```

Comment:
Is this done instead of `source`ing the venv, or for other reasons?

Reply:
Yes it is done because sourcing needs to set ENV (PATH) and would only persist for the current command.

Reply:
Ah I see. This demystifies it for me https://pythonspeed.com/articles/activate-virtualenv-dockerfile/